### PR TITLE
TST: reset rcparams in animation smoke test

### LINKED ISSUE #6523: matplotlib-2.0.0b1 test errors on Windows
matplotlib-2.0.0b1 on 64-bit Python 3.5 for Windows errors on two tests. 
It looks like [updating the rcParams at runtime](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/animation.py#L667) with the full path to ImageMagick's convert tool no longer has any effect during the tests (it still works outside the tests suite).
```
======================================================================
ERROR: __main__.test_save_animation_smoketest('imagemagick_file', 'gif')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python35\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "X:\Python35\lib\site-packages\matplotlib\testing\decorators.py", line 152, in wrapped_callable
    func(*args, **kwargs)
  File "test_animation.py", line 57, in check_save_animation
    anim.save(F.name, fps=30, writer=writer, bitrate=500)
  File "X:\Python35\lib\site-packages\matplotlib\animation.py", line 838, in save
    writer.grab_frame(**savefig_kwargs)
  File "X:\Python35\lib\contextlib.py", line 66, in __exit__
    next(self.gen)
  File "X:\Python35\lib\site-packages\matplotlib\animation.py", line 202, in saving
    self.finish()
  File "X:\Python35\lib\site-packages\matplotlib\animation.py", line 402, in finish
    + ' Try running with --verbose-debug')
RuntimeError: Error creating movie, return code: 4 Try running with --verbose-debug

======================================================================
ERROR: __main__.test_save_animation_smoketest('imagemagick', 'gif')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python35\lib\site-packages\matplotlib\animation.py", line 236, in grab_frame
    dpi=self.dpi, **savefig_kwargs)
  File "X:\Python35\lib\site-packages\matplotlib\figure.py", line 1533, in savefig
    self.canvas.print_figure(*args, **kwargs)
  File "X:\Python35\lib\site-packages\matplotlib\backend_bases.py", line 2245, in print_figure
    **kwargs)
  File "X:\Python35\lib\site-packages\matplotlib\backends\backend_agg.py", line 508, in print_raw
    fileobj.write(renderer._renderer.buffer_rgba())
OSError: [Errno 22] Invalid argument

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "X:\Python35\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "X:\Python35\lib\site-packages\matplotlib\testing\decorators.py", line 152, in wrapped_callable
    func(*args, **kwargs)
  File "test_animation.py", line 57, in check_save_animation
    anim.save(F.name, fps=30, writer=writer, bitrate=500)
  File "X:\Python35\lib\site-packages\matplotlib\animation.py", line 838, in save
    writer.grab_frame(**savefig_kwargs)
  File "X:\Python35\lib\site-packages\matplotlib\animation.py", line 244, in grab_frame
    'with --verbose-debug.'.format(e, out, err))
OSError: Error saving animation to file (cause: [Errno 22] Invalid argument) Stdout: b'' StdError: b'Invalid Parameter - 800x600\r\n'. It may help to re-run with --verbose-debug.

```

### PR DESCRIPTION
**Pull request by tacaswell**
Closes #6523 

attn @cgohlke

### CODE REVIEW

**Comment by QuLogic** [comment]
An explanatory comment would not be unwelcome.

**Comment by tacaswell** [comment]
@QuLogic fixed.

File: lib/matplotlib/tests/test_animation.py [hunk 1]
```diff
@@ -30,6 +30,13 @@ def test_save_animation_smoketest():
 
 @cleanup
 def check_save_animation(writer, extension='mp4'):
+    try:
+        # for ImageMagick the rcparams must be patched to account for
+        # 'concvert' being a built in MS tool, not the imagemagick
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_animation.py]: convert

**Comment by QuLogic** [comment]
I'd also mention that it's because `cleanup` reset the rcparams; sometimes people read the tests for ideas, and maybe we don't want them assuming this is necessary.

**Comment by tacaswell** [comment]
This maybe revealing a much larger problem that this fix is sweeping under the rug.  Some one needs to look into if `mpl.style.use` is also resetting this.  I suspect in the animation module we need to reach in and touch the defaults.

**Pull request merged**